geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyGatewayReceiverCommand.java

```diff
@@ -0,0 +1,95 @@
+import org.apache.geode.management.internal.cli.i18n.CliStrings;
+import org.apache.geode.management.internal.cli.result.CommandResult;
+import org.apache.geode.management.internal.cli.result.ResultBuilder;
+import org.apache.geode.management.internal.security.ResourceOperation;
+import org.apache.geode.security.ResourcePermission;
+
+public class DestroyGatewayReceiverCommand extends InternalGfshCommand {
+  public static final String DESTROY_GATEWAYRECEIVER = "destroy gateway-receiver";
+  public static final String DESTROY_GATEWAYRECEIVER__HELP =
+      "Destroy the Gateway Receiver on a member or members.";
+  public static final String DESTROY_GATEWAYRECEIVER__GROUP__HELP =
+      "Group(s) of members on which to destroy the Gateway Receiver.";
+  public static final String DESTROY_GATEWAYRECEIVER__MEMBER__HELP =
+      "Name/Id of the member on which to destroy the Gateway Receiver.";
+  public static final String GATEWAYRECIEVER_NOT_FOUND_ON_0 = "Gateway Receiver not found on {0}";
+
+  @CliCommand(value = DESTROY_GATEWAYRECEIVER, help = DESTROY_GATEWAYRECEIVER__HELP)
+  @CliMetaData(relatedTopic = CliStrings.TOPIC_GEODE_WAN)
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.MANAGE, target = ResourcePermission.Target.GATEWAY)
+  public Result destroyGatewayReceiver(
+      @CliOption(key = {CliStrings.GROUP, CliStrings.GROUPS},
+          optionContext = ConverterHint.MEMBERGROUP,
+          help = DESTROY_GATEWAYRECEIVER__GROUP__HELP) String[] onGroups,
+      @CliOption(key = {CliStrings.MEMBER, CliStrings.MEMBERS},
+          optionContext = ConverterHint.MEMBERIDNAME,
+          help = DESTROY_GATEWAYRECEIVER__MEMBER__HELP) String[] onMember,
+      @CliOption(key = CliStrings.IFEXISTS, help = CliStrings.IFEXISTS_HELP,
+          specifiedDefaultValue = "true", unspecifiedDefaultValue = "false") boolean ifExists) {
+
+    boolean persisted = false;
+    Set<DistributedMember> members = getMembers(onGroups, onMember);
+
+    List<CliFunctionResult> functionResults =
+        executeAndGetFunctionResult(DestroyGatewayReceiverFunction.INSTANCE, ifExists, members);
+
+    CommandResult result = ResultBuilder.buildResult(functionResults);
+
+    // Only update the cluster config if the command is not executed on specific members.
+    InternalClusterConfigurationService service =
+        (InternalClusterConfigurationService) getConfigurationService();
+    if (onMember == null && result.getStatus().equals(Result.Status.OK) && service != null) {
+      if (onGroups == null) {
+        onGroups = new String[] {"cluster"};
+      }
+      Arrays.stream(onGroups).forEach(group -> service.updateCacheConfig(group, cc -> {
+        CacheConfig.GatewayReceiver receiver = cc.getGatewayReceiver();
+        if (receiver == null) {
+          throw new EntityNotFoundException(
+              CliStrings.format(GATEWAYRECIEVER_NOT_FOUND_ON_0, group), ifExists);
```

Comment:
Will this not cause the loop to exit early and skip other groups?

Reply:
Good catch, I'd noticed that earlier but forgot to fix it.

For the fix, I've dispensed with throwing the exception and set the overall command status to ERROR (taking in to account `ifExists`) if a group config doesn't exist. Also required a change `InternalClusterConfigurationService` to better handling of update of a noexistent group config. No more NPE.